dev/fattest.simplicity/src/componenttest/annotation/SkipForRepeat.java

```diff
@@ -26,13 +30,77 @@
 @Retention(RetentionPolicy.RUNTIME)
 public @interface SkipForRepeat {
 
+    String[] value();
+
     public static final String NO_MODIFICATION = EmptyAction.ID;
     public static final String EE7_FEATURES = EE7FeatureReplacementAction.ID;
     public static final String EE8_FEATURES = EE8FeatureReplacementAction.ID;
     public static final String EE9_FEATURES = JakartaEEAction.EE9_ACTION_ID;
     public static final String EE10_FEATURES = JakartaEEAction.EE10_ACTION_ID;
     public static final String EE11_FEATURES = JakartaEEAction.EE11_ACTION_ID;
 
-    String[] value();
+    // Cannot use MultivalueSkips.name() since it isn't a constant at compile time so need to actually
```

Comment:
Is there a way of separating the EE specific parts from the generic SkipForRepeat annotation? Ideally I would have removed the EE constants from this file completely as they exist elsewhere (e.g. JakartaEEAction).

We will probably want to add similar MP specific code and I wouldn't want to put it in this class.

Reply:
I actually like the constants here because it is easy to find the constants when using the SkipForRepeat annotation on a method / class.  If we were to move them from this class, it would be better to be in a new class called RepeatActionConstants and could have all of the constants for EE and MP and not be in their own separate EE and MP repeat action classes.

I limited this PR so that people could start using this new function to accelerate their EE 11 work and start updating FATs.

Reply:
The constants should go next to wherever the `MultiValueRepeat` enum is. At the moment that's in the `SkipForRepeat` annotation.

However, I don't like this change in general. It's not going to work for anything except EE repeats and it's not readily extensible to microprofile tests (where there's a much longer list of repeat IDs) or FAT buckets which define their own repeats.

I've two main objections to the existing `@SkipForRepeat` annotation:
1. It's usually going to be clearer to say which repeats a test applies to rather than which it should be skipped for
2. It should really be extensible so that we can define new conditions we haven't thought of yet
3. Ideally we should be able to say things like "I only want this test to run if the repeat has "myFeature-2.0" or later", rather than having to list the specific repeats

I understand that making something extensible is going to take longer so we may need this as a stopgap in the short term, but we're very quickly going to have similar problems that this approach doesn't readily solve.

Reply:
That is all well and good.  I am just trying to provide the requested function that was identified by a person doing EE 11 development in order to improve productivity of adding new EE versions.  It sounds like you are wanting to have additional annotations like `@RepeatOnly` and possibly something else for the additional conditions you are asking for.  Annotations can only take compile time constants so we cannot have things like Condition and Predicates to satisfy annotations.  There is Assume.assumeThat() that can be done in tests as well to provide some additional conditions and not use annotations at all.

Reply:
I would put the MultiValueRepeat into the RepeatActionConstants class if I moved these constants to their own class of course.

Reply:
> I am just trying to provide the requested function that was identified by a person doing EE 11 development in order to improve productivity of adding new EE versions.

That is all well and good, but we are going to have more requirements in this area shortly.

On the other hand, this doesn't make it any more difficult to fix the problem properly further down the line, so I don't object to it for the moment.